src/action_creators/tile.js

```diff
@@ -1,10 +1,29 @@
 import {ActionTypes} from '../constants';
 
-export function selectTile(tileId) {
+function selectTile(tileinfo) {
+  if (tileinfo.player) {
+    tileinfo.cellstate = '';
+    return selectRackTile(tileinfo);
+  }
+  tileinfo.selected = {row: Math.floor(tileinfo.selected / 15), col: tileinfo.selected % 15};
```

Comment:
so, this will change the object by reference.  I don't suggest doing that because it's pretty hard to debug when you run into a problem, and you want to figure out why something is wrong.  

this is akin to changing a class definition (by adding a couple of properties) in a function in Java.  you can't do it in Java, and you shouldn't do it here :)